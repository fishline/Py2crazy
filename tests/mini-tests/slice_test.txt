a[x + y]
a[x + y:foo(x, y)]
a[x + y:foo(x, y):-(3+5)]
a[x + y,...,-(3+5)]
