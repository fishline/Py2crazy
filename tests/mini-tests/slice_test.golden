Disassembling top-level module in ./mini-tests/slice_test.txt
  1       0 LOAD_NAME            [43m[30ma[39m[49m[x + y]
          3 LOAD_NAME            a[[43m[30mx[39m[49m + y]
          6 LOAD_NAME            a[x + [43m[30my[39m[49m]
          9 BINARY_ADD           a[[43m[30mx + y[39m[49m]
         10 BINARY_SUBSCR        a[[43m[30mx + y[39m[49m]
         11 POP_TOP              a[[43m[30mx + y[39m[49m]

  2      12 LOAD_NAME            [43m[30ma[39m[49m[x + y:foo(x, y)]
         15 LOAD_NAME            a[[43m[30mx[39m[49m + y:foo(x, y)]
         18 LOAD_NAME            a[x + [43m[30my[39m[49m:foo(x, y)]
         21 BINARY_ADD           a[[43m[30mx + y[39m[49m:foo(x, y)]
         22 LOAD_NAME            a[x + y:[43m[30mfoo[39m[49m(x, y)]
         25 LOAD_NAME            a[x + y:foo([43m[30mx[39m[49m, y)]
         28 LOAD_NAME            a[x + y:foo(x, [43m[30my[39m[49m)]
         31 CALL_FUNCTION        a[x + y:[43m[30mfoo(x, y)[39m[49m]
         34 SLICE+3              a[x + y:[43m[30mfoo(x, y)[39m[49m]
         35 POP_TOP              a[x + y:[43m[30mfoo(x, y)[39m[49m]

  3      36 LOAD_NAME            [43m[30ma[39m[49m[x + y:foo(x, y):-(3+5)]
         39 LOAD_NAME            a[[43m[30mx[39m[49m + y:foo(x, y):-(3+5)]
         42 LOAD_NAME            a[x + [43m[30my[39m[49m:foo(x, y):-(3+5)]
         45 BINARY_ADD           a[[43m[30mx + y[39m[49m:foo(x, y):-(3+5)]
         46 LOAD_NAME            a[x + y:[43m[30mfoo[39m[49m(x, y):-(3+5)]
         49 LOAD_NAME            a[x + y:foo([43m[30mx[39m[49m, y):-(3+5)]
         52 LOAD_NAME            a[x + y:foo(x, [43m[30my[39m[49m):-(3+5)]
         55 CALL_FUNCTION        a[x + y:[43m[30mfoo(x, y)[39m[49m:-(3+5)]
         58 LOAD_CONST           a[x + y:foo(x, y):-([43m[30m3[39m[49m+5)]
         61 LOAD_CONST           a[x + y:foo(x, y):-(3+[43m[30m5[39m[49m)]
         64 BINARY_ADD           a[x + y:foo(x, y):-([43m[30m3+5[39m[49m)]
         65 UNARY_NEGATIVE       a[x + y:foo(x, y):[43m[30m-(3+5[39m[49m)]
         66 BUILD_SLICE          a[x + y:foo(x, y):[43m[30m-(3+5[39m[49m)]
         69 BINARY_SUBSCR        a[x + y:foo(x, y):[43m[30m-(3+5[39m[49m)]
         70 POP_TOP              a[x + y:foo(x, y):[43m[30m-(3+5[39m[49m)]

  4      71 LOAD_NAME            [43m[30ma[39m[49m[x + y,...,-(3+5)]
         74 LOAD_NAME            a[[43m[30mx[39m[49m + y,...,-(3+5)]
         77 LOAD_NAME            a[x + [43m[30my[39m[49m,...,-(3+5)]
         80 BINARY_ADD           a[[43m[30mx + y[39m[49m,...,-(3+5)]
         81 LOAD_CONST           a[[43m[30mx + y[39m[49m,...,-(3+5)]
         84 LOAD_CONST           a[x + y,...,-([43m[30m3[39m[49m+5)]
         87 LOAD_CONST           a[x + y,...,-(3+[43m[30m5[39m[49m)]
         90 BINARY_ADD           a[x + y,...,-([43m[30m3+5[39m[49m)]
         91 UNARY_NEGATIVE       a[x + y,...,[43m[30m-(3+5[39m[49m)]
         92 BUILD_TUPLE          a[x + y,...,[43m[30m-(3+5[39m[49m)]
         95 BINARY_SUBSCR        a[x + y,...,[43m[30m-(3+5[39m[49m)]
         96 POP_TOP              a[x + y,...,[43m[30m-(3+5[39m[49m)]
         97 LOAD_CONST           a[x + y,...,[43m[30m-(3+5[39m[49m)]
        100 RETURN_VALUE         a[x + y,...,[43m[30m-(3+5[39m[49m)]
